### ISSUE TITLE
RetroArch search does not work

### ISSUE THREAD

**Issue by AlanRibas** [libretro/RetroArch#12349]
I have currently installed 53 different systems on RetroArch.
However, I can't find the games I have using RetroArch's search engine.
When I perform a search for any name, which I know from my games, one or the other name appears as a result, but not even close to what corresponds to reality.
Is it my impression or is this research field only looking for games related to the MAME emulator?

<img width="960" alt="A" src="https://user-images.githubusercontent.com/16063712/116819795-fff1da80-ab47-11eb-9de9-37cbd1220274.png">

**Comment by dezraj** [comment]
I'm experiencing the same behavior. I've downloaded the portable (zip) version of 1.9.2 and unzipped it to a new, clean folder. When I look for content in my computer, nothing is found in my directories, even though they are the same ones I use in my previous version of Retroarch (v. 1.9.0).

Search engine seems to have been broken somehow.

**Comment by AlanRibas** [comment]
People also wanted to make a suggestion regarding this search tab in RetroArch.
You could add the possibility to enable / disable these types of more specific searches:
Example:
- By developer.
- Per annum.
- Launch.
- System name.

In my opinion, the most important search mode is the one already present at the top of the search screen. Where you add the name of the game you want and click on ENTER to see the result.

IN the name of God ..... that a human being when searching for a game would say "well, I don't remember the name of the game but I remember it was made in 1995, so I will look for it by date"

**Comment by hizzlekizzle** [comment]
The 'explore' search is less about finding a single game that you already know everything about, since just going to a specific playlist and searching there is faster and more precise anyway. It's more for game discovery. For example, "show me all the Mario games... Oh, I didn't even realize *these* existed," or "I really like this game that came out in 1995, show me some other games from the same year", etc.

**Comment by AlanRibas** [comment]
> The 'explore' search is less about finding a single game that you already know everything about, since just going to a specific playlist and searching there is faster and more precise anyway. It's more for game discovery. For example, "show me all the Mario games... Oh, I didn't even realize _these_ existed," or "I really like this game that came out in 1995, show me some other games from the same year", etc.

I understood a proposal of this tool.
But in my opinion it would be interesting to leave it up to the user to decide how to use it. Adding the possibility of a general search, on all installed systems. And also the possibility to disable these search modes that will not be used.

For example, when I go to play with some friends, we always searched by the name someone spoke (Using a search engine for a frontend called GameEx).
The search mode you mentioned may be just one of the modes. However, it is currently the only one. But as I said before, it seems to be having a problem.

**Comment by gouchi** [comment]
Please be specific about the issue. 

The function is an **Explore feature** from the content matched from Libretro database.

If you did manual scan, your content won't appear in search name from Explore feature.

Thank you.

**Comment by AlanRibas** [comment]
The problem is:
Then type any game name in the search tab and the results that appear are only for those games belonging to MAME.

**Comment by hizzlekizzle** [comment]
The pause is known and technically unavoidable. Dunno why you're only getting MAME stuff. Are your other games scanned in using the database scanner (i.e., *not* manual scanning)?

**Comment by AlanRibas** [comment]
> The pause is known and technically unavoidable. Dunno why you're only getting MAME stuff. Are your other games scanned in using the database scanner (i.e., _not_ manual scanning)?

All of my games were added to RetroArch through the desktop menu of that frontend. I currently have 53 different systems installed on RetroArch. The only difference between them all is that the MAME system is: The MAME system was added through that option called:
Import Content / Manual Analysis
Within that option above I added an .XML file from MamePlus2013.

All other systems were added with me dragging their roms to the specific location of the emulator in the Desktop area of the RetroArch frontend.

I don't know if I managed to be clear in my answer.

**Comment by hizzlekizzle** [comment]
Yes, that explains it, then. The 'explore' search requires games to be imported through the database scanning. Playlists created through the manual scan or desktop menu's drag-and-drop skip over the database checks, thereby making the playlists ineligible for the explore function.

**Comment by AlanRibas** [comment]
> Yes, that explains it, then. The 'explore' search requires games to be imported through the database scanning. Playlists created through the manual scan or desktop menu's drag-and-drop skip over the database checks, thereby making the playlists ineligible for the explore function.

but the use of .XML lists changes the names of the games when used.  Right?  I for example rename each of my roms from all systems ... (yes ... very stupid ... it was a while ago and I am not proud of that ... well ... maybe a little).  With this edition I wanted to leave the names without that lot of unnecessary information .. like USA, EU, JAP, (!), Apart from other lots of information that sometimes a rom brings.  But anyway, wouldn't it be possible to change the retroarch to scan the list produced by the user?  Unless that user adds an XML file for the system in question?

**Comment by dezraj** [comment]
So this issue seems more like a feature request than a real bug. I'm opening another issue then because my case is different. I come from a clean (zip) install of version 1.9.2 and when I go over load content -> playlists -> import content -> scan directory, point to a directory to be scanned, no content is found, therefore no playlists are created. This doesn't happen in version 1.9.0.

**Comment by AlanRibas** [comment]
> So this issue seems more like a feature request than a real bug. I'm opening another issue then (#12366) because my case is different. I come from a clean (zip) install of version 1.9.2 and when I go over load content -> playlists -> import content -> scan directory, point to a directory to be scanned, no content is found, therefore no playlists are created. This doesn't happen in version 1.9.0.

I understood. I actually thought the tool's behavior was in trouble. But after clarifying our colleagues here, I realized that it works that way.
In my case, then, can I leave this post here and serve as an implementation request? Or should I open a new post in a different place?

**Comment by hizzlekizzle** [comment]
Sure, you can leave it. Maybe just change the title to reflect, like '[feature request] sort by tags' or something like that.

This was the original idea for the "cursor" system, btw, it just never really took off, as the cursors have to be pre-created instead of on-the-fly.

**Comment by AlanRibas** [comment]
My request for implantation is:

The possibility that once the roms were inserted in RetroArch they can be searched using the "Search" tool.

And if it is still possible, that the additional options for refining the search can be enabled and activated depending on the user's interest (In this way, this option is less polluted with options that specific users are not interested in using.).

**Comment by LibretroAdmin** [comment]
It's not only for mame games, not sure where that confusion came from.
